Lee

LeeTKho opened the pull request:
added 2 notebooks:
- one notebook does historical tweet searches based on location and saves results to csv
- other notebook for preliminary keyword analysis

LeeTKho committed:
Create Historical Tweet Search.ipynb

LeeTKho committed:
removed keyword analysis stuff

LeeTKho committed:
moved keyword analysis to separate notebook

christaina commented:
can you add .py versions of your notebooks as well? easier to leave comments that way. you can make a .py version of a notebook by running `jupyter nbconvert --to python notebook_name.ipynb` in the terminal from the folder the nb is in

LeeTKho committed:
added .py files

LeeTKho commented:
done!

christaina commented on Historical Tweet Search.py:
```diff
@@ -0,0 +1,36 @@
+    c.Geo = search[0]
+    c.Output = "historical_search_"+search[1]+".csv"
+    c.Search = "blm OR \"black lives matter\" OR police OR protest OR riot OR \"breonna taylor\" OR \"george floyd\"" #"blm" #search terms
+    #c.Min_likes = 20
```
did you comment it out bc there was something wrong w/ that config or just for your own testing??

LeeTKho replied:
i was originally trying different search terms so that's why there are those comments, i will remove them to avoid confusion

christaina commented on Keyword Analysis.py:
```diff
@@ -0,0 +1,87 @@
+# In[4]:
+
+
+df_BK.iloc[:,1:].sum(axis=0).sort_values(ascending=False)[:50]
```
seems pretty nice barclays was at the top, since its pretty specific. was there a protest there??

LeeTKho replied:
yah so the location i set for brooklyn tweets was centered around grand army plaza and included barclays center; i chose this location because that was where the largest protests in brooklyn were happening so i thought it might have more relevant tweet activity

christaina commented on Keyword Analysis.ipynb:
```diff
@@ -0,0 +1,245 @@
+    {
+     "data": {
+      "text/plain": [
+       "police                306\n",
```
interesting police appears 100 more times in SI over brooklyn. did you check out any of the tweets w/ the word police in them to see if there are any differences in the content between bk and si?

LeeTKho replied:
haven't taken a closer look at the tweets themselves, i can check that out today

LeeTKho replied:
turns out the SI search pulled in over twice as many tweets as the BK search, so that's probably why "police" showed up way more in SI. i think i'll add a tfidf analysis to the the keyword search notebook so that the stats will (hopefully) be more comparable between data sets.

christaina commented on Keyword Analysis.ipynb:
```diff
@@ -0,0 +1,245 @@
+      "text/plain": [
+       "police                306\n",
+       "twitter               197\n",
+       "twitter com           191\n",
```
looks like all these top terms are coming in from people having links in their tweets. can we add them as custom excluded words?

LeeTKho replied:
sure! i can work on that

LeeTKho committed:
added tfidf analysis to keyword analysis notebook and added code to put tweets into pandas df; also added pickled tweet dfs

LeeTKho committed:
adding hashtags network graph notebook and updating .py files

LeeTKho committed:
added D3 graph and json file

LeeTKho committed:
updates

added pickled tweet files for all boroughs, basic graphs for all boroughs and unfinished clustered graphs for mn and bk. also made some changes to notebooks

LeeTKho committed:
updates
